building-applications/expression-language.md

```diff
@@ -0,0 +1,73 @@
+| trim(input)                       | Returns the input string with all leading and trailing spaces removed. If the input is not a string, it attempts a string conversion. |
+| unpack(input, fieldsList)         | Returns a map containing the elements of `input`, for each field in the `fieldList` you will see an entry in the map. If the input is a string it is converted to a list using the `split()` function with the '`,`' separator |
+| split(input, separatorExpression) | Split the input to a list of strings, this is internally using the String.split() function. An empty input corresponds to an empty list. The input is convered to a String using the str() function. |
+| timestampAdd(input, delta, unit)  | Returns a timestamp formed by adding `delta` in `unit` to the `input` timestamp. `input` is a timestamp to add to. `delta` is a `long` amount of `unit` to add to `input`. Can be a negative value to perform subtraction. `unit` the string unit of time to add or subtract. Can be one of [`years`, `months`, `days`, `hours`, `minutes`, `seconds`, `millis`]. |
+
+
+## Conditional Steps
+
+Each `step` accepts an optional `when` configuration that is evaluated at step execution time against current records (the current step in the transform pipeline).
+
+\
+The `when` condition supports the expression language syntax, which provides access to the record attributes as follows:
+
+| **Name (field)**  | **Description**                                                                                                                    |
+| ----------------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| key:              | the key portion of the record in a KeyValue schema.                                                                                |
+| value:            | the value portion of the record in a KeyValue schema, or the message payload itself.                                               |
+| messageKey:       | the optional key messages are tagged with (aka. Partition Key).                                                                    |
+| topicName:        | the optional name of the topic which the record originated from (aka. Input Topic).                                                |
+| destinationTopic: | the name of the topic on which the transformed record will be sent (aka. Output Topic).                                            |
+| eventTime:        | the optional timestamp attached to the record from its source. For example, the original timestamp attached to the pulsar message. |
+| properties:       | the optional user-defined properties attached to record.                                                                           |
+
+You can use the `.` operator to access top level or nested properties on a schema-full key or value.\
+For example, `key.keyField1` or `value.valueFiled1.nestedValueField`.
+
```

Comment:
We have to add another special "field reference" that is "record".
You can (must) use "record" in order to refer to the "current record" in a loop, this is currently used in:
- the fn:filter function
- the compute-ai-embeddings agent when you use the "loop-over" feature (see "Flare" and the ref for the compute-ai-embedding agent)
- the query (query-vector-db) agent when you use the "loop-over" feature (see "Flare" and the ref for the query/query-vector-db agent)